### PR TITLE
Publish more data for imageDetails via DockerV2 task

### PR DESCRIPTION
**Pull request by dabasnidhi**
Publishing more information in ImageDetails Service -
JobName, ImageSize, LayerSize, Layers Created Date and renamed build to pipeline in all parameters.

For Testing - http://nidabas-desk/DefaultCollection/p1/_serviceEndpoints/09b3aa23-0c50-4958-9d59-5f5e42ec364d/kubernetesSummary?namespace=kube-system

image

### CODE REVIEW

File: Tasks/Common/docker-common/dockercommandutils.ts [hunk 1]
```diff
@@ -91,26 +91,39 @@ export async function getLayers(connection: ContainerConnection, imageId: string
 
 function parseHistory(input: string) {
     const NOP = '#(nop)';
-    var directive = 'UNSPECIFIED';
-    var argument = '';
-    var index = input.indexOf(NOP);
+    let directive = "UNSPECIFIED";
+    let argument = "";
+    let index: number = input.indexOf(NOP);
+    const createdByMatch = "; createdBy:";
```
Comment (madhuv-msft) [inline Tasks/Common/docker-common/dockercommandutils.ts]: are we okay with this, as this is not going to be localized.

File: Tasks/DockerV2/dockerpush.ts [hunk 2]
```diff
@@ -103,17 +108,19 @@ export function run(connection: ContainerConnection, outputUpdate: (data: string
...
-     const layers = await dockerCommandUtils.getLayers(connection, imageName);
+    const layers = await dockerCommandUtils.getLayers(connection, imageName);
 
     // Getting pipeline variables
-    const buildId = parseInt(tl.getVariable("Build.BuildId"));
-    const buildDefinitionName = tl.getVariable("Build.DefinitionName");
-    const buildVersion = tl.getVariable("Build.BuildNumber");
-    const buildDefinitionId = tl.getVariable("System.DefinitionId");
+    const hostType = tl.getVariable("System.HostType").toLowerCase();
+    const runId = hostType === "build" ? parseInt(tl.getVariable("Build.BuildId")) : parseInt(tl.getVariable("Release.ReleaseId"));
```
Comment (madhuv-msft) [inline Tasks/DockerV2/dockerpush.ts]: build can be const and also it can be read from a common variable.

File: Tasks/DockerV2/task.loc.json [hunk 3]
```diff
@@ -14,7 +14,7 @@
   "version": {
     "Major": 2,
     "Minor": 150,
-    "Patch": 4
+    "Patch": 5
```
Comment (madhuv-msft) [inline Tasks/DockerV2/task.loc.json]: why are we not changing for task.json of dockerv2?

File: Tasks/Common/docker-common/dockercommandutils.ts (hunk 1 above)
Comment (dabasnidhi) [reply Tasks/Common/docker-common/dockercommandutils.ts]: We are setting these same as the Docker history command arguments.  Also this data will be visible in logs only when system debug is true, So I think we need not localise

File: Tasks/DockerV2/task.loc.json (hunk 3 above)
Comment (dabasnidhi) [reply Tasks/DockerV2/task.loc.json]: Task.json for DockerV2 is currently at patch 5, whereas for task.loc.json it was set to 4. I have now updated both to 6

File: Tasks/DockerV2/dockerpush.ts (hunk 2 above)
Comment (dabasnidhi) [reply Tasks/DockerV2/dockerpush.ts]: Made the string build a const. Also for pipeline name I changed to a common variable, system.DefinitionName. But I did not find any common var for build and release Ids
image

**Pull request merged**